classes/class-p4-campaign-exporter.php

```diff
@@ -60,18 +59,18 @@ public function single_post_export_bulk_action() {
 			$wp_list_table   = _get_list_table( 'WP_Posts_List_Table' );
 			$action          = $wp_list_table->current_action();
 			$allowed_actions = array( 'export' );
-			if ( ! in_array( $action, $allowed_actions ) ) {
+			if ( ! in_array( $action, $allowed_actions, true ) ) {
 				return false;
 			}
 			switch ( $action ) {
 				case 'export':
-					$sendback = 'admin.php?action=export_data&post=' . join( ',', $_REQUEST['post'] );
+					$sendback = 'admin.php?action=export_data&post=' . join( ',', array_map( 'absint', $_REQUEST['post'] ) );
```

Comment:
should use `filter_input()` here as well

Reply:
I tried it but the `post` param is a array and filter_input function isn't provide the filtered array data.

Reply:
In this case I think it might be better to use [array_filter](https://php.net/manual/en/function.array-filter.php) to validate each one of the post ids and exclude those who are not valid integers.
Can you try something like this?
```
$validated_posts = array_filter(
	$_REQUEST['post'],
	function ( $element ) {
		return filter_var( $element, FILTER_VALIDATE_INT );
	}
);
```

Reply:
looks perfect 👍